src/pkg/services/m365/api/mail.go

```diff
@@ -285,106 +285,179 @@ func (c Mail) GetItem(
 		return mail, MailInfo(mail, size), nil
 	}
 
-	attachConfig := &users.ItemMessagesItemAttachmentsRequestBuilderGetRequestConfiguration{
-		QueryParameters: &users.ItemMessagesItemAttachmentsRequestBuilderGetQueryParameters{
-			Expand: []string{"microsoft.graph.itemattachment/item"},
-		},
-		Headers: newPreferHeaders(preferPageSize(maxNonDeltaPageSize), preferImmutableIDs(immutableIDs)),
+	attachments, totalSize, err := c.getAttachments(ctx, userID, mailID, immutableIDs)
+	if err != nil {
+		// A failure can be caused by having a lot of attachments.
+		// If that happens, we can progres with a two-step approach of:
+		// 1. getting all attachment IDs.
+		// 2. fetching each attachment individually.
+		logger.CtxErr(ctx, err).Info("falling back to fetching attachments by id")
+
+		attachments, totalSize, err = c.getAttachmentsIterated(ctx, userID, mailID, immutableIDs, errs)
+		if err != nil {
+			return nil, nil, clues.Stack(err)
+		}
 	}
 
-	attached, err := c.LargeItem.
+	size += totalSize
```

Comment:
nit: We can lose this and just use `totalSize`.

Reply:
not quite; see line 280.  we need to add in the size of the message body first.

Reply:
Ah, thanks!